src/net/tls.cc

```diff
@@ -339,6 +339,18 @@ struct gnutls_datum : public gnutls_datum_t {
         data = nullptr;
         size = 0;
     }
+    gnutls_datum(const gnutls_datum&) = delete;
+    gnutls_datum& operator=(gnutls_datum&& other) {
+        if (this == &other) {
+            return *this;
+        }
+        if (data != nullptr) {
+            ::gnutls_free(data);
+        }
+        data = std::exchange(other.data, nullptr);
+        size = std::exchange(other.size, 0);
+        return *this;
+    }
```

Comment:
The type is never moved. The warning is somewhat aggressive, since it only addresses a potential move. I suggest simply adding a delete for both move and copy constructors. Should never be needed.

Reply:
@elcallio hi Calle, thank you for your review. do you imply that at https://github.com/scylladb/seastar/blob/a62c9dbc9219b41ebeeab822b2389c56d7638648/src/net/tls.cc#L425 , `_session_resume_key` is always empty?

Reply:
No, sorry, missed/forgot that assignment. You are of course right.

Reply:
@elcallio thank =) may i consider your reply an approval of this change?

Reply:
Yes, though I think perhaps you could get the same effect by just defining a move constructor, no? (Fewer lines).

Reply:
A move constructor won't help, this is an assignment.